tests/test_elf.py

```diff
@@ -0,0 +1,144 @@
+import sys
+import subprocess
+import re
+import os
+import shutil
+from komodo import elf
+
+
+# NOTE: If these tests are failing, know that they're meant for AMD64 Linux
+# machines only. If we're testing komodo on a different CPU architecture or OS
+# kernel, these will need to be updated.
+
+
+def _compile(args, code="int main() {}\n"):
+    with open("main.c", "w") as f:
+        f.write(code)
+    assert subprocess.call(re.split(r"\s+", args.format(code="main.c"))) == 0
+
+
+def _make_executable(dest):
+    _compile("gcc -o{} {{code}}".format(dest))
+
+
+def _make_shared_object(dest):
+    _compile("gcc -shared -fPIC -o{} {{code}}".format(dest))
+
+
+def _make_text_file(dest):
+    with open(dest, "w") as f:
+        f.write("#!/usr/bin/python\nprint('Hello\n')")
+
+
+def test_exec():
+    """Check whether Python is a valid executable (Hint: it is)"""
+    assert elf.is_valid_elf_file(sys.executable)
+
+
+def test_dyn(tmpdir):
+    """Compile a shared object and test it"""
+    with tmpdir.as_cwd():
+        _compile("gcc -shared {code}")
+
+        assert os.path.isfile("a.out")
+        assert elf.is_valid_elf_file("a.out")
+
+
+def test_symlink(tmpdir):
```

Comment:
Just to be certain, should we make a test that symlinks a folder, and assert that the files inside that folder are not counted as valid files?

Reply:
Yep! See enclosed commit.

Reply:
Alright, if the tests are green I have nothing against building a test release :+1: